lib/ethclient/ethclient.go

```diff
@@ -96,20 +97,18 @@ func (w Wrapper) HeaderByType(ctx context.Context, typ HeadType) (*types.Header,
 	const endpoint = "header_by_type"
 	defer latency(w.chain, endpoint)()
 
-	var header *types.Header
-	err := w.cl.Client().CallContext(
-		ctx,
-		&header,
-		"eth_getBlockByNumber",
-		typ.String(),
-		false,
-	)
+	var bn rpc.BlockNumber
+	if err := bn.UnmarshalJSON([]byte(typ.String())); err != nil {
+		return nil, errors.Wrap(err, "unmarshal head type")
+	}
+
+	header, err := w.cl.HeaderByNumber(ctx, big.NewInt(int64(bn)))
 	if err != nil {
 		incError(w.chain, endpoint)
-		return nil, errors.Wrap(err, "get block")
+		err = errors.Wrap(err, "json-rpc", "endpoint", endpoint)
```

Comment:
following the pattern in the generated wrappers for consistency

Reply:
nice